### PR TITLE
Share UI Submit fix

### PR DESCRIPTION
**Pull request by SofiaSazonova**
### Feature or Bugfix

- Bugfix

### Detail
- Submit button only available for (Requesters' || 'ApproversAndRequesters')
- Only  (Requesters' || 'ApproversAndRequesters') can edit request purpose

### Relates
- <URL or Ticket>

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: frontend/src/modules/Shared/Shares/ShareEditForm.js [hunk 1]
```diff
@@ -221,15 +221,28 @@ export const ShareEditForm = (props) => {
...
+
   const handlePageChange = async (event, value) => {
     if (value <= sharedItems.pages && value !== sharedItems.page) {
       await setFilter({ ...filter, page: value });
     }
   };
 
   const beforeClose = async () => {
-    if (smthChanged || requestPurpose !== share.requestPurpose) {
+    if (requestPurpose !== share.requestPurpose) {
```
Comment (noah-paige) [inline frontend/src/modules/Shared/Shares/ShareEditForm.js:242]: need to a `setSmthChanged(true);` for if no response errors for `updateRequestPurpose` then

Comment (SofiaSazonova) [reply frontend/src/modules/Shared/Shares/ShareEditForm.js:242]: updateRequestPurpose doesn't change the state of Share. So, basically, there is nothing to submit.

Comment (SofiaSazonova) [reply frontend/src/modules/Shared/Shares/ShareEditForm.js:242]: @noah-paige may be we should change that, but not before 2.6

Comment (noah-paige) [reply frontend/src/modules/Shared/Shares/ShareEditForm.js:242]: `onApply` just refreshed the ShareObject and associated share items in the view

If we do not run `onApply()` then I can update the request purpose --> hit `Save & Close` --> the share view is not updated and the old request purpose still shows

**Review by noah-paige — APPROVED** [review]
lgtm

**Pull request merged**